src/battle_main.c

```diff
@@ -3677,6 +3677,13 @@ static void TryDoEventsBeforeFirstTurn(void)
         if (ItemBattleEffects(ITEMEFFECT_ON_SWITCH_IN, gBattlerByTurnOrder[gBattleStruct->switchInItemsCounter++], FALSE))
             return;
     }
+    
+    if (B_FIRST_BATTLE_TUTORIAL == TRUE && gBattleTypeFlags & BATTLE_TYPE_FIRST_BATTLE && !gBattleStruct->tutorialDone)
```

Comment:
IMO, config constants should be used to surround blocks of code instead of being used as actual checks on the code itself. I'm gonna post a PR that fixes this for most of them.